shardingsphere-features/shardingsphere-encrypt/shardingsphere-encrypt-core/src/main/java/org/apache/shardingsphere/encrypt/rule/EncryptTable.java

```diff
@@ -187,6 +187,16 @@ public Optional<Boolean> getQueryWithCipherColumn() {
         return Optional.ofNullable(queryWithCipherColumn);
     }
     
+    /**
+     * Get query with cipher column.
+     *
+     * @param logicColumn logic column
+     * @return query with cipher column
+     */
+    public Optional<Boolean> getQueryWithCipherColumn(final String logicColumn) {
```

Comment:
How about inherit the isQueryWithCipherColumn on table with column here?

Reply:
yes,  It can be

```java
    public Optional<Boolean> getQueryWithCipherColumn(final String logicColumn) {
        return Optional.ofNullable(findEncryptColumn(logicColumn).map(EncryptColumn::getQueryWithCipherColumn).orElse(queryWithCipherColumn));
    }
```